tests/test_2022_03_24.py

```diff
@@ -0,0 +1,234 @@
+
+    ##
+    ## START VOTE
+    ##
+    vote_id = vote_id_from_env or start_vote({'from': ldo_holder}, silent=True)[0]
+
+    tx: TransactionReceipt = helpers.execute_vote(
+        vote_id=vote_id, accounts=accounts, dao_voting=dao_voting
+    )
+
+    factories_after = easy_track.getEVMScriptFactories()
+
+    rewards_manager_balance_after = ldo_token.balanceOf(one_inch_reward_manager)
+    hr_multisig_balance_after = ldo_token.balanceOf(hyperelliptic_rockx_multisig)
+    dao_balance_after = ldo_token.balanceOf(dao_agent_address)
+
+    assert hr_multisig_balance_after - hr_multisig_balance_before == hyperelliptic_rockx_comp.amount
+    assert rewards_manager_balance_after == 0
+
+    assert dao_balance_before - dao_balance_after == \
+        hyperelliptic_rockx_comp.amount + (rewards_manager_balance_after - rewards_manager_balance_before)
+
+    for f in factories_to_add_with_vote:
+        assert f.factory_addr in factories_after
+
+    _check_factories_sanity(accounts[5], accounts[6], ldo_token, easy_track)
+
+    ### validate vote events
+    # assert count_vote_items_by_events(tx) == 6, "Incorrect voting items count"
+
+    display_voting_events(tx)
+
+    if bypass_events_decoding:
+        return
+
+    evs = group_voting_events(tx)
+
+    # asserts on vote item 1
+    validate_evmscript_factory_added_event(evs[0], factories_to_add_with_vote[0])
+
+    # asserts on vote item 2
+    validate_evmscript_factory_added_event(evs[1], factories_to_add_with_vote[1])
+
+    # asserts on vote item 3
+    validate_evmscript_factory_added_event(evs[2], factories_to_add_with_vote[2])
+
+    # asserts on vote item 4
+    validate_payout_event(evs[3], hyperelliptic_rockx_comp)
+
+def test_2022_03_24_zero_ldo_to_recover(
+    helpers, accounts, ldo_holder, dao_voting, ldo_token,
+    vote_id_from_env, bypass_events_decoding, easy_track,
+    dao_agent
+):
+    assert dao_agent.address == dao_agent_address
+
+    rewards_manager_balance_before = ldo_token.balanceOf(one_inch_reward_manager)
+    rewards_manager = interface.RewardsManager(one_inch_reward_manager)
+    dao_balance_before = ldo_token.balanceOf(dao_agent_address)
+
+    # if rewards manager has no tokens transfer it from agent
+    if rewards_manager_balance_before > 0:
+        rewards_manager.recover_erc20(ldo_token, rewards_manager_balance_before, {"from": dao_agent})
+
+    assert ldo_token.balanceOf(one_inch_reward_manager) == 0
+
+    ##
+    ## START VOTE
+    ##
+    vote_id = vote_id_from_env or start_vote({'from': ldo_holder}, silent=True)[0]
+
+    tx: TransactionReceipt = helpers.execute_vote(
+        vote_id=vote_id, accounts=accounts, dao_voting=dao_voting
+    )
+
+    rewards_manager_balance_after = ldo_token.balanceOf(one_inch_reward_manager)
+    dao_balance_after = ldo_token.balanceOf(dao_agent_address)
+
+    assert dao_balance_before - dao_balance_after == hyperelliptic_rockx_comp.amount - rewards_manager_balance_before
+    assert rewards_manager_balance_after == 0
+
```

Comment:
I think we also need to check the owner of the rewards manager here.

Reply:
added, thx!